Mathlib/RepresentationTheory/Coinvariants.lean

```diff
@@ -0,0 +1,182 @@
+/-
+Copyright (c) 2025 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+import Mathlib.RepresentationTheory.Rep
+
+/-!
+# Coinvariants a group representation
+
+Given a commutative ring `k` and a monoid `G`, this file introduces the coinvariants of a
+`k`-linear `G`-representation `(V, ρ)`.
+
+We first define `Representation.augmentationSubmodule`, the submodule of `V` generated by elements
+of the form `ρ g x - x` for `x : V`, `g : G`. Then the coinvariants of `(V, ρ)` are the quotient of
+`V` by this submodule. We show that the functor sending a representation to its coinvariants is
+left adjoint to the functor equipping a module with the trivial representation.
+
+-/
+
+universe u v
+
+namespace Representation
+
+variable {k G V W : Type*} [CommRing k] [Monoid G] [AddCommGroup V] [Module k V]
+  [AddCommGroup W] [Module k W] (ρ : Representation k G V) (τ : Representation k G W)
+
+/-- The submodule of a representation generated by elements of the form `ρ g x - x`. -/
+def augmentationSubmodule : Submodule k V :=
+  Submodule.span k (Set.range fun (x : G × V) => ρ x.1 x.2 - x.2)
+
+variable {ρ}
+
+lemma mem_augmentationSubmodule_of_eq (g : G) (x : V) (a : V) (h : ρ g x - x = a) :
+    a ∈ augmentationSubmodule ρ :=
+  Submodule.subset_span ⟨(g, x), h⟩
+
+variable (ρ)
+
+/-- The coinvariants of a representation, `V ⧸ ⟨{ρ g x - x | g ∈ G, x ∈ V}⟩`. -/
+def Coinvariants := V ⧸ augmentationSubmodule ρ
```

Comment:
Are you sure you don't just want `abbrev`? If not, can you add to the docstring where the API diverges from the quotient type?